### ISSUE TITLE
Allow specifying multiple areas and areas by % using -a option

### ISSUE THREAD

**Issue by asheeshrana** [tabulapdf/tabula-java#209]
From commandline, currently we can only specify one area at a time. I want to specify multiple areas as well as areas using top, left, bottom, right as % of height and width of the page for my use case.

Multiple areas are helpful when trying to extract tables from multiple parts of the page using commandline

Specifying areas using % of height and width is useful when I don't want to calculate absolute value of top/left/bottom/right points to define areas.

I will be happy to do the changes and create pull request myself if such a change is acceptable.

**Comment by jazzido** [comment]
Hi @asheeshrana,

That's a good suggestion, and should be pretty easy to implement. My only concern is keeping the command line options consistent, and easy to understand.

BTW, we'd love to get a pull request for this feature :)

**Comment by asheeshrana** [comment]
Thanks for the response.
I propose to change the behavior of "-a" option as follows:
-a top,left,bottom,right,top,left,bottom,right...

Each group of 4 parameters would define an area that could be used to extract the table

If the value of parameters top, left, bottom, right is between 0-1 (inclusive), we can consider it as a fractions of height and width... e.g. the below parameters
-a 0,0,1,0.5,0,0.5,1,1 would define two areas such that
area 1 = (top = 0*height, left = 0*width, bottom = 1*height, right = 0.5*width)
area 2 = (top = 0*height, left = 0.5*width, bottom = 1*height, right = 1*width)
Basically it is defining areas that split page into two areas right at the middle.

And update the description for -a option as
-a/--area = Portion of the page to analyze. Accepts sets of (top,left,bottom,right). Example: --area 269.875,12.75,790.5,561,300,570,500,700. Additionally, if all values are between 0-1 (inclusive), input will be taken as fraction of actual height or width of the page. Example: --area %0,0,1,0.5,0,0.5,1,1. Default is entire page

Let me know if you have any concerns... Expect a pull request soon :).

**Comment by criztovyl** [comment]
I have concerns about the between-0-and-1 overloading of `--area`, for me a parameter should not accept ambiguous formats.
I would rather go by explicitly specifying that you want percent/fractions.

I would suggest using a prefix, in example `%` and `/` (fraction): `--area %0,0,100,50,50,100,100` and `--area /0,0,1,0.5,0.5,1,1`.

`--fraction` and `--percent(-age)` might also be possible, but `-p` is already `--pages`, don't know for `-f`. :D

**Comment by asheeshrana** [comment]
~~~
I would suggest using a prefix, in example % and / (fraction): --area %0,0,100,50,50,100,100 and --area /0,0,1,0.5,0.5,1,1.
~~~

@criztovyl 
I will take your suggestion and use / to explicitly indicate that it is a fraction. Using both / and % seems redundant so I will only allow "/". 

New description
-a/--area = Portion of the page to analyze. Accepts sets of (top,left,bottom,right). Example: --area 269.875,12.75,790.5,561,300,570,500,700. Additionally, if all values are between 0-1 (inclusive) and preceded by "/", input will be taken as fraction of actual height or width of the page. Example: --area /0,0,1,0.5,0,0.5,1,1. Default is entire page

PS: -f is being used for format (JSON, CSV etc...)

**Comment by jazzido** [comment]
+1 to the prefix.

BTW, let's make sure that we're validating the input. 

Usability-wise, percentages (0-100) strike me better than 0-1 fractional values. What do you think?

**Comment by asheeshrana** [comment]
I don't have preference for either. I will do % then.

New description
-a/--area = Portion of the page to analyze. Accepts sets of (top,left,bottom,right). Example: --area 269.875,12.75,790.5,561,300,570,500,700. Additionally, if all values are between 0-100 (inclusive) and preceded by "%", input will be taken as % of actual height or width of the page. Example: --area %0,0,100,50,50,100,100. Default is entire page

**Comment by jazzido** [comment]
BTW, I'm not sure of extending the valid inputs for `-a` so it accepts _sets_ of rectangular areas.

What about changing `CommandLineApp` so it allows repeated appearances of the `-a` argument? Your example would be something like:

```
-a 269.875,12.75,790.5,561 - a 300,570,500,700.
```

**Comment by asheeshrana** [comment]
I agree multiple occurrences of-a is much cleaner.

Updated description...
-a/--area = Portion of the page to analyze. Accepts top,left,bottom,right . Example: --area 269.875,12.75,790.5,561. If all values are between 0-100 (inclusive) and preceded by "%", input will be taken as % of actual height or width of the page. Example: --area %0,0,100,50. To specify multiple areas, -a option should be repeated. Default is entire page

**Comment by asheeshrana** [comment]
Created the pull request... [tabula-java allow specifying multiple areas
](https://github.com/tabulapdf/tabula-java/pull/213)

**Comment by asheeshrana** [comment]
The PR was merged and hence closing this issue.
